lib/OnyxUtils.ts

```diff
@@ -804,6 +804,9 @@ function keyChanged<TKey extends OnyxKey>(
             return;
         }
     }
+
+    const cachedCollections: Record<string, ReturnType<typeof getCachedCollection>> = {};
```

Comment:
`cachedCollections` stores all collections all in one object? Let's verify that we have enough memory for that 😅

Reply:
Haha :D

On serious note, it won't store all the collections, only those from `subscriber.key`. Also the reference is hold for the `keyChange` lifespan and the object size is the same as we've got from `getCachedCollection`. 

Furthermore, it will hold a reference to only one of such collection object, whereas multiple calls to `getCachedCollection` spawn many of those. 

Let me know if my understanding is correct here, thanks!